core/lib/dal/migrations/20240611121747_add_base_token_price_table.up.sql

```diff
@@ -0,0 +1,11 @@
+CREATE TABLE base_token_prices (
+    id SERIAL PRIMARY KEY,
+    created_at TIMESTAMP NOT NULL,
+    updated_at TIMESTAMP NOT NULL,
+
+    ratio_timestamp TIMESTAMP NOT NULL,
```

Comment:
why do we need the timestamp?

From the DAL, I don't see any use case where we need the base_token_price other than the latest.

The table will grow infinitely and fetching price will get slower over time

Reply:
To the very least we're going to need to persist the 
- Latest ratio
- Latest ratio used in the L1
Besides that, we were thinking that multiple records might be needed for debugging.

We were planning on adding a cleaning process for this table. Our plan was to roll it out without one, see how this is used, and create one accordingly. Perhaps we should already go ahead and persist only some days worth of history.

Reply:
gotcha

I wonder if keeping the record is necessary as it will be persisted on L1 anyways ?

The latest L1 ratio can also be fetched on L1, no need to duplicate it in DB.

I think only the boolean `used_in_l1` really matter here so that we know if the component crash midway, then we need to start by sending to L1 before fetching the new rate

Reply:
Why we persist the ratio:
- If the external price APIs crash, you don't want the server to halt
- Be able to know what the last ratio stored on the L1 and be prudent with frequent L1 updating
- Possible debugging
- For the operator to potentially look back and collect data on prices to analyze their off-chain BaseToken<->ETH conversions

Also, there's a ticket to add to `HouseKeeper` the ability to clear rows over 7 days old.